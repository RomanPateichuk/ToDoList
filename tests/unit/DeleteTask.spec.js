import { mount } from "@vue/test-utils";
import DeleteTaskComponent from "../../src/components/atoms/RemoveTask.vue";
import { mutations } from "../../src/store/index.js";

describe("Тестирование компонента DeleteTask:", () => {
  const wrapper = mount(DeleteTaskComponent, {
    propsData: {},
  });
  test("соответствие снимку", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Должен удалять задачу из Store", () => {
    const state = {
      tasks: [
        { id: "1", value: "Task 1", checked: false },
        { id: "2", value: "Task 2", checked: false },
      ],
    };

    mutations.deleteTask(state, "1");
    expect(state).toEqual({
      tasks: [{ id: "2", value: "Task 2", checked: false }],
    });
  });
});
