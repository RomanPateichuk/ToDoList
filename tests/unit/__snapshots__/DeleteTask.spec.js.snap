// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента DeleteTask: соответствие снимку 1`] = `<button>×</button>`;
